src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java

```diff
@@ -313,6 +344,26 @@ public static double readDouble(final long p_ptr) {
         return MS_UNSAFE_HANDLER.getUnsafe().getDouble(p_ptr);
     }
 
+    /**
+     * Write an array of booleans to memory.
+     *
+     * @param p_ptr
+     *         Start address to write to.
+     * @param p_array
+     *         Array with data to write.
+     * @param p_arrayOffset
+     *         Offset in array to start reading the data from.
+     * @param p_length
+     *         Number of elements to write.
+     * @return Number of written elements
+     */
+    public static int writeBooleans(final long p_ptr, final boolean[] p_array, final int p_arrayOffset, final int p_length) {
+        MS_UNSAFE_HANDLER.getUnsafe().copyMemory(p_array, UnsafeHandler.getArrayByteOffset() + p_arrayOffset, null,
```

Comment:
The base offset is not intended for boolean arrays. Please add a new method (`getArrayBooleanOffset`), which returns the appropriate value `sun.misc.Unsafe.ARRAY_BOOLEAN_BASE_OFFSET`.

Reply:
Done